spec/std/pointer_spec.cr

```diff
@@ -231,6 +231,14 @@ describe "Pointer" do
     a[2].should eq(5)
   end
 
+  it "maps_with_index!, with offset" do
+    a = Pointer(Int32).malloc(3) { |i| i + 1 }
+    a.map_with_index!(3, offset: 10) { |e, i| e + i }
+    a[0].should eq(11)
```

Comment:
you could probably do what you did above in https://github.com/crystal-lang/crystal/pull/8264/files#diff-fbe3f630f4ca2da753ea76cf3a0b95baR1594

```suggestion
    a.should eq([11, 13, 15])
```